docs/understanding-airbyte/airbyte-protocol.md

```diff
@@ -108,7 +108,7 @@ The `spec` command allows an actor to broadcast information about itself and how
 check(Config) -> AirbyteConnectionStatus
 ```
 
-The `check` command validates that, given a configuration, that the Actor is able to connect and access all resources that it needs in order to operate. e.g. Given some Postgres credentials, it determines whether it can connect to the Postgres database. If it can, it will return a success response. If it fails (perhaps the password is incorrect), it will return a failed response and (when possible) a helpful error message. If an actor's `check` command succeeds, it is expected that all subsequent methods in the sync will also succeed.
+The `check` command validates that, given a configuration, that the Actor is able to connect and access all resources that it needs in order to operate. e.g. Given some Postgres credentials, it determines whether it can connect to the Postgres database. If it can, it will return a success response. If it fails because of a configuration issue (perhaps the password is incorrect), it will return a failed response and (when possible) a helpful error message. A failed response will be considered as a config error. If it fails because of a connector issue, the `check` command should return a trace message. If an actor's `check` command succeeds, it is expected that all subsequent methods in the sync will also succeed.
```

Comment:
```suggestion
The `check` command validates that, given a configuration, that the Actor is able to connect and access all resources that it needs in order to operate. e.g. Given some Postgres credentials, it determines whether it can connect to the Postgres database. The output will be as follows:
- If it can, the `check` command will return a success response. 
- If `check` fails because of a configuration issue (perhaps the password is incorrect), it will return a failed response and (when possible) a helpful error message. A failed response will be considered as a config error, i.e. user error. Outputting a trace message detailing the config error is optional, but allows for more detailed debugging of the error.
- If it fails because of a connector issue, the `check` command should output a trace message detailing the failure. It is not expected to receive an `AirbyteConnectionStatus` in this failure case. 

If an actor's `check` command succeeds, it is expected that all subsequent methods in the sync will also succeed.
```